lib/explorer/backend/lazy_series.ex

```diff
@@ -1151,6 +1155,43 @@ defmodule Explorer.Backend.LazySeries do
     Backend.Series.new(data, {:u, 32})
   end
 
+  @impl true
+  def count_matches(series, substring) do
+    data = new(:count_matches, [lazy_series!(series), substring], {:u, 32})
+
+    Backend.Series.new(data, {:u, 32})
+  end
+
+  @impl true
+  def re_count_matches(series, pattern) do
+    data = new(:re_count_matches, [lazy_series!(series), pattern], {:u, 32})
+
+    Backend.Series.new(data, {:u, 32})
+  end
+
+  @impl true
+  def re_scan(series, pattern) do
+    data = new(:re_scan, [lazy_series!(series), pattern], {:list, :string})
+
+    Backend.Series.new(data, {:list, :string})
+  end
+
+  @impl true
+  def re_named_captures(_series, _pattern) do
+    raise """
+    #{unsupported(:re_named_captures, 2)}
+
+    If you want to capture named groups from a column, you must do so outside of a query.
+    For example, instead of:
+
+        Explorer.DataFrame.mutate(df, new_column: re_named_captures(column, ~S/(a|b)/))
+
+    You must write:
+
+        Explorer.DataFrame.put(df, :new_column, Explorer.Series.re_named_captures(column, ~S/(a|b)/))
+    """
```

Comment:
We could probably make this work by storing the original backend as a field of the the LazySeries and then asking it to return the names. But we can do this in another PR if desired, no worries for now IMO.

Reply:
I don't know yet how to capture that information in the context of the lazy frame, but I will research a bit. For now I'm going to merge this. Thanks!